### ISSUE TITLE
Let users change username

### ISSUE THREAD

**Issue by guaka** [Trustroots/trustroots#595]
Currently these are "handled" manually via support.

Users should be able to change username via Account settings.

It should be limited to only once every 3 months so we'll have to transfer 

Ideally, we'd also store their previous usernames in some kind of array stored in the user model, but that's a stretch goal.

**Comment by guaka** [comment]
(It could also be just functionality to be used by volunteers, only to be used if people have a good reason to change their username.
Either way, I don't have a strong opinion, but we need to have something more than a bunch of DB queries:)

**Comment by simison** [comment]
@trmendes would you be interested in start working on this?

I think it could be split into two PRs to ensure it's easy to work on and easy to review:

- API changes in "update user"
- Frontend changes

**Comment by trmendes** [comment]
I'm short on time here but I would definitely try it. \o/ Do you mind helping me if I have questions? (irc?)

**Comment by trmendes** [comment]
@guaka I have a question here: what is the use in saving/store the changes a user made on his/her 'usernames'? I see you put it as a "would be nice to have feature" but I just want to understand better what do you have in mind and discuss it :)

**Comment by simison** [comment]
@trmendes storing username history is:
- allows redirects
- allows not letting someone else register with that username
- allows us to find reported users who might've just changed username

I wrote it down (I edited the initial message to include that) just as a note — we shouldn't include that in the first implementation.

First step should be:
- Add a new field to record date when the username was changed
- If username field is updated, compare to that date field and if it's newer than 3 months, return failure.

**Comment by simison** [comment]
> So I made a few changes here and it is working but I still have a few questions.

You can open a work-in-progress pull request with questions and I can reply there. :-)

**Comment by mrkvon** [comment]
It seems that if

>  - allows redirects
>  - allows not letting someone else register with that username
>  - allows us to find reported users who might've just changed username

is going to be implemented, the feature should not go to production without

>  - allows not letting someone else register with that username

Perhaps obvious, but I wanted to have it noted. :)

**Comment by trmendes** [comment]
I believe I forgot to check it the username is already in use. Going to have a check and fix it if if it is need...thanks for the note @mrkvon

**Comment by trmendes** [comment]
#617 -> @simison @mrkvon 

How should I proceed with this one? Because to implement the history I need to make this patch OK first then I can use this branch to work on the next version which will have the code to implement the issue #617
